src/programs/candyMachine/gpaBuilders/CandyMachineGpaBuilder.ts

```diff
@@ -0,0 +1,29 @@
+import { GpaBuilder } from '@/utils';
+import { PublicKey } from '@solana/web3.js';
+
+type AccountDiscriminator = [number, number, number, number, number, number, number, number];
+// TODO(thlorenz): copied from candy machine SDK
+// SDK should either provide a GPA builder or expose this discriminator
+const candyMachineDiscriminator: AccountDiscriminator = [51, 173, 177, 113, 25, 241, 109, 189];
+
+const AUTHORITY = candyMachineDiscriminator.length;
+const WALLET = AUTHORITY + PublicKey.default.toBytes().byteLength;
+
+export class CandyMachineGpaBuilder extends GpaBuilder {
+  whereDiscriminator(discrimator: AccountDiscriminator) {
+    return this.where(0, Buffer.from(discrimator));
+  }
+
+  candyMachineAccounts() {
+    return this.whereDiscriminator(candyMachineDiscriminator);
+  }
+
+  // wallet same as solTreasury
+  candyMachineAccountsForWallet(wallet: PublicKey) {
```

Comment:
Remind me to have a chat with you about this. I have recently been creating GpaBuilder classes for each account type (that extend a program GpaBuild when needed) where the constructor enforces the account constraint. I think this is good for now but something to discuss when we think about auto-generating them.